File: devsh/exercises/spark-streaming/stubs-python/StreamingLogs.py
import sys
from pyspark import SparkContext
from pyspark.streaming import StreamingContext

# Given an RDD of KB requests, print out the count of elements
def printRDDcount(t, rdd): print t, ":", "Number of KB requests: "+str(rdd.count())

if __name__ == "__main__":
    if len(sys.argv) != 3:
        print >> sys.stderr, "Usage: StreamingLogs.py <hostname> <port>"
        sys.exit(-1)
    
    # get hostname and port of data source from application arguments
    hostname = sys.argv[1]
    port = int(sys.argv[2])
     
    # Create a new SparkContext
    sc = SparkContext()

    # Set log level to ERROR to avoid distracting extra output
    sc.setLogLevel("ERROR")

    # TODO  
    ssc = StreamingContext(sc, 1)
    DStream = ssc.socketTextStream(hostname, port)\
        .filter(lambda line:"KBDOC" in line)

    DStream.pprint(5)
    DStream.foreachRDD(lambda t,r: printRDDcount(t, r))
    
    ssc.start()
    ssc.awaitTermination()

File: devsh/exercises/spark-streaming/streaminglogs_project/src/main/scala/stubs/StreamingLogs.scala
package stubs

import org.apache.spark.SparkContext
import org.apache.spark.streaming.StreamingContext
import org.apache.spark.streaming.Seconds

object StreamingLogs {
  def main(args: Array[String]) {
    if (args.length < 2) {
      System.err.println("Usage: stubs.StreamingLogs <hostname> <port>")
      System.exit(1)
    } 
 
    // get hostname and port of data source from application arguments
    val hostname = args(0)
    val port = args(1).toInt
    
    // Create a new SparkContext
    val sc = new SparkContext()

    // Set log level to ERROR to avoid distracting extra output
    sc.setLogLevel("ERROR")
            
    // TODO
    println("Stub not yet implemented")

  }
}
